src/main/java/frc/robot/constants/ArmConstants.java

```diff
@@ -0,0 +1,110 @@
+package frc.robot.constants;
+
+import edu.wpi.first.math.util.Units;
+
+/** Add your docs here. */
+public class ArmConstants {
+
+  private ArmConstants() {
+    throw new IllegalStateException("Constants Class");
+  }
+
+  public enum ArmState {
+    UNDEFINED(10000, 10000), STOW(MIN_PIV_ANGLE_RAD, MIN_EXT_LEN_IN),
+
+    LOWER_NODE_CONE(Units.degreesToRadians(30), 0), // TBD
+    LOWER_NODE_CUBE(Units.degreesToRadians(30), 0), // TBD
+
+    MID_NODE_CONE(Units.degreesToRadians(66.05), 0), // Done
+    MID_NODE_CUBE(Units.degreesToRadians(87.0), 0), // TBD
+
+    UPPER_NODE_CONE(Units.degreesToRadians(88.6), 12.98), // EXT Calulated
+    UPPER_NODE_CUBE(Units.degreesToRadians(100), 17.1), // TBD
+
+    SUBSTATION_CONE(Units.degreesToRadians(85.19), 0), // TBD
+    SUBSTATION_CUBE(Units.degreesToRadians(90), 0), // TBD
+
+    GROUND_PICKUP_CONE(Units.degreesToRadians(47), 16.7), // Calculated
+    // 125 cm away from center
+    GROUND_PICKUP_CUBE(Units.degreesToRadians(40), 9.5); // Not popssible
+
+    public final double pivotAngleRad;
+    public final double extentionDistIn;
+
+    ArmState(double piv, double ext) {
+      pivotAngleRad = piv;
+      extentionDistIn = ext;
+    }
+  }
+
+  // Common Configurables
+  public static final boolean RATE_LIMIT_ARM = false;
+  public static final boolean ENABLE_FEEDFORWARD = false;
+  public static final double KG = 0.03;
+
+  // channel on dio port
+  public static final int DIOPORTPIV = 0;
+  public static final int CONNECTION_THRESH_HZ = 975;
+
+
+  public static final double PIVOT_ABS_ENC_TO_ROTATION = 1.0 / 3.7142857;
+  public static final double EXTENSION_ROTATION_TO_INCHES = 18.5 / 7.01;
+
+
+  // Offset For Pivot
+  public static final double PIV_INIT_OFFSET_ROT = .032 + 0.0633;
```

Comment:
In the very rare event that the arm slips during competitions, software technicians can easily reset it before a match. However, being able to reconfigure the arm offset during a match can get slightly risky and might mess up a lot, especially considering the high-stress environment where anything can happen. Would it be wise to put that up to the operator, rather than just resetting it before a match?

Reply:
Instead of a incrementer for the offset, a reset encoders button might work better.

Reply:
Resetting extension encoders would be a valid way to combat slippage. The only thing we would have to worry about is the pivot, which rarely has slippage(unless serious damage occurs) and can be easily fixed outside of a match. The only real benefit configurable offsets have is being able to fix it within a match, which is a capability that I would argue the operator shouldn't have because of the high-stress environment and the fact that configurable offsets can easily break the arm mechanically(like what happened in girls gen). On top of this, they would have <2 minutes in teleop to fix this by which point time would be over.